operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcher.java

```diff
@@ -103,20 +105,65 @@ public boolean matches(R actual, R desired, Context<?> context) {
 
     removeIrrelevantValues(desiredMap);
 
-    if (LoggingUtils.isNotSensitiveResource(desired)) {
-      logDiff(prunedActual, desiredMap, objectMapper);
+    var matches = prunedActual.equals(desiredMap);
+
+    if (!matches && LoggingUtils.isNotSensitiveResource(desired)) {
+      var diff = getDiff(prunedActual, desiredMap, objectMapper);
+      if (log.isDebugEnabled()) {
+        log.debug(
+            "Diff between actual and desired state for resource: {} with name: {} in namespace: {} is: \n{}",
+            actual.getKind(), actual.getMetadata().getName(), actual.getMetadata().getNamespace(),
+            diff);
+      }
     }
 
-    return prunedActual.equals(desiredMap);
+    return matches;
   }
 
-  private void logDiff(Map<String, Object> prunedActualMap, Map<String, Object> desiredMap,
+  private String getDiff(Map<String, Object> prunedActualMap, Map<String, Object> desiredMap,
       KubernetesSerialization serialization) {
-    if (log.isDebugEnabled()) {
-      var actualYaml = serialization.asYaml(prunedActualMap);
-      var desiredYaml = serialization.asYaml(desiredMap);
-      log.debug("Pruned actual yaml: \n {} \n desired yaml: \n {} ", actualYaml, desiredYaml);
```

Comment:
We still might want to log the actual resources.

Reply:
when should we log actual resource?

Reply:
I mean it is good that  we log the diff, but some might be interested in the actual resources (I mean both "actual" and desired yaml). It might be useful to log that too?

Reply:
> What do you think?

how about logging actual resource in case when  logger is enabled for the trace level? or should we introduce new configuration parameter?

Reply:
Trace level is fine by me, we could have there both the resources.

Reply:
ok, I enabled printing actual and desired resources when trace is enabled.